x-pack/plugins/security/server/routes/licensed_route_handler.ts

```diff
@@ -4,10 +4,22 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { RequestHandler } from 'kibana/server';
+import { KibanaResponseFactory, RequestHandler, RouteMethod } from 'kibana/server';
 
-export const createLicensedRouteHandler = <P, Q, B>(handler: RequestHandler<P, Q, B>) => {
-  const licensedRouteHandler: RequestHandler<P, Q, B> = (context, request, responseToolkit) => {
+export const createLicensedRouteHandler = <
+  P,
+  Q,
+  B,
+  M extends RouteMethod,
+  R extends KibanaResponseFactory
+>(
+  handler: RequestHandler<P, Q, B, M, R>
+) => {
+  const licensedRouteHandler: RequestHandler<P, Q, B, M, R> = (
```

Comment:
To be honest I have mixed feelings about this since feature that depends on license level can rely on multiple endpoints and keeping all of them in sync may be tedious. Unless we refactor `SecurityLicense` to include something similar to `feature <-> min license level` mapping that will be the only source of truth.

Reply:
That’s a great point, a single source of truth is much more beneficial than the potential duplication of license checks